lib/std/jsbigints.nim

```diff
@@ -0,0 +1,159 @@
+  runnableExamples:
+    doAssert (newJsBigInt"9" + newJsBigInt"1") == newJsBigInt"10"
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"9" - newJsBigInt"1") == newJsBigInt"8"
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"42" * newJsBigInt"9") == newJsBigInt"378"
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".} =
+  ## Same as `div` but for `JsBigInt`(uses JavaScript `BigInt() / BigInt()`).
+  runnableExamples:
+    doAssert newJsBigInt"13" div newJsBigInt"3" == newJsBigInt"4"
+    doAssert newJsBigInt"-13" div newJsBigInt"3" == newJsBigInt"-4"
+    doAssert newJsBigInt"13" div newJsBigInt"-3" == newJsBigInt"-4"
+    doAssert newJsBigInt"-13" div newJsBigInt"-3" == newJsBigInt"4"
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".} =
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+  runnableExamples:
+    doAssert newJsBigInt"13" mod newJsBigInt"3" == newJsBigInt"1"
+    doAssert newJsBigInt"-13" mod newJsBigInt"3" == newJsBigInt"-1"
+    doAssert newJsBigInt"13" mod newJsBigInt"-3" == newJsBigInt"1"
+    doAssert newJsBigInt"-13" mod newJsBigInt"-3" == newJsBigInt"-1"
+
+func `<`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert newJsBigInt"2" < newJsBigInt"9"
+
+func `<=`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".} =
+  runnableExamples:
+    doAssert newJsBigInt"1" <= newJsBigInt"5"
+
+func `==`*(x, y: JsBigInt): bool {.importjs: "(# === #)".} =
+  runnableExamples:
+    doAssert newJsBigInt"42" == newJsBigInt"42"
+
+func `**`*(x, y: JsBigInt): JsBigInt {.importjs: "((#) $1 #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"9" ** newJsBigInt"5") == newJsBigInt"59049"
+
+func `xor`*(x, y: JsBigInt): JsBigInt {.importjs: "(# ^ #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"555" xor newJsBigInt"2") == newJsBigInt"553"
+
+func `shl`*(a, b: JsBigInt): JsBigInt {.importjs: "(# << #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"999" shl newJsBigInt"2") == newJsBigInt"3996"
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".} =
+  runnableExamples:
+    doAssert (newJsBigInt"999" shr newJsBigInt"2") == newJsBigInt"249"
+
+func `-`*(a: JsBigInt): JsBigInt {.importjs: "($1#)".} =
+  runnableExamples:
+    doAssert -(newJsBigInt"10101010101") == newJsBigInt"-10101010101"
+
+func inc*(a: var JsBigInt; b = 1) {.importjs: "([#][0][0] += BigInt(#))".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    inc big1, 2
+    doAssert big1 == newJsBigInt"3"
+
+func dec*(a: var JsBigInt; b = 1) {.importjs: "([#][0][0] -= BigInt(#))".} =
+  runnableExamples:
+    var big1: JsBigInt = newJsBigInt"1"
+    dec big1, 2
+    doAssert big1 == newJsBigInt"-1"
+
+func `+=`*(x: var JsBigInt; y: int) {.importjs: "([#][0][0] $1 BigInt(#))".} =
```

Comment:
this doesn't make sense. Not only does it mask the fact that `y` gets converted to BigInt, but also it prevents simple things like this:
```
var a = newJsBigInt"1234"
a += a
```

It's also inconsistent to have `JsBigInt+JsBigInt` but JsBigInt+=int)

The previous way was better: `func `+=`*(x: var JsBigInt; y: JsBigInt)`. 
these procs need to change to the previous way, using only JsBigInt arguments:
* inc, dec, +=, -=, *=, /=

Reply:
https://github.com/nim-lang/Nim/pull/16409#discussion_r549345897
https://github.com/nim-lang/Nim/pull/16409#discussion_r549345937
:shrug:

Reply:
sorry for the conflicting reviewer opinions... there are arguments for either camp (but += definitely should be `BigInt, BigInt`), we can debate this in a linked RFC maybe